### PR TITLE
Rewrite the service layer in typescript.

### PR DESCRIPTION
**Pull request by AAJELLAL**

### CODE REVIEW

File: src/components/network/network-map-tab.tsx [hunk 1]
```diff
@@ -506,6 +506,7 @@ export const NetworkMapTab = ({
                 `Loading geo data of study '${studyUuid}' of missing substations '${notFoundSubstationIds}' and missing lines '${notFoundLineIds}'...`
             );
             dispatch(setMapDataLoading(true));
+            console.log(' notFoundSubstationIds: ', notFoundSubstationIds);
```
Comment (jamal-khey) [inline src/components/network/network-map-tab.tsx:509]: console.log to remove

File: src/services/directory.ts [hunk 2]
```diff
@@ -6,10 +6,11 @@
  */
 
 import { backendFetchJson, getRequestParamFromList } from './utils';
+import { UnknownArray } from 'type-fest';
 
 const PREFIX_DIRECTORY_SERVER_QUERIES = import.meta.env.VITE_API_GATEWAY + '/directory';
 
-export function fetchContingencyAndFiltersLists(listIds) {
+export function fetchContingencyAndFiltersLists(listIds: UnknownArray) {
```
Comment (jamal-khey) [inline src/services/directory.ts:13]: UUID [] ? or at least a string[]

File: src/services/network-modification-types.ts [hunk 3]
```diff
@@ -37,3 +48,307 @@ export interface VscModificationInfo {
     connectionPosition?: string | null;
     terminalConnected?: boolean | null;
 }
+
+type AttributeModification<T> = { value: T; op: string };
```
Comment (jamal-khey) [inline src/services/network-modification-types.ts:52]: this type is already defined in converter-station-utils.tsx. may be you can remove one of the definitions ?

File: src/services/network-modification-types.ts [hunk 4]
```diff
@@ -37,3 +48,307 @@ export interface VscModificationInfo {
...
+    connectionPosition?: string | null;
+    terminalConnected?: boolean | null;
+    properties?: Property[];
+}
+
+export interface StaticVarCompensatorCreationInfo {
+    studyUuid: string;
+    nodeUuid: UUID;
+    staticCompensatorId: string;
+    staticCompensatorName: string | null | undefined;
```
Comment (jamal-khey) [inline src/services/network-modification-types.ts:166]: to be more concistent with the remaning definitions in this class
```suggestion
    staticCompensatorName?: string | null;
```

File: src/services/network-modification-types.ts [hunk 5]
```diff
@@ -37,3 +48,307 @@ export interface VscModificationInfo {
...
+    equipmentId: string;
+    reactiveCapabilityCurvePoints:
+        | {
+              p: number | null;
+              oldP: number | null;
+              minQ: number | null;
+              oldMinQ: number | null;
+              maxQ: number | null;
+              oldMaxQ: number | null;
+          }[]
```
Comment (jamal-khey) [inline src/services/network-modification-types.ts:339]: create a a type in converter-station-utils.tsx ?

File: src/services/network-modification-types.ts [hunk 6]
```diff
@@ -37,3 +48,307 @@ export interface VscModificationInfo {
...
+    maxQ: AttributeModification<number> | null;
+}
+
+export interface Assignment {
+    dataType: DataType | undefined;
+    value: FieldValue | undefined;
+    filters: Filter[];
+    editedField: string;
+    propertyName?: string | undefined;
+}
```
Comment (jamal-khey) [inline src/services/network-modification-types.ts:354]:
```suggestion
export interface Assignment {
    dataType?: DataType;
    value?: FieldValue;
    filters: Filter[];
    editedField: string;
    propertyName?: string;
}
```

File: src/services/study/geo-data.ts [hunk 7]
```diff
@@ -7,8 +7,13 @@
 
 import { backendFetchJson, getQueryParamsList } from '../utils';
 import { getStudyUrlWithNodeUuid } from './index';
+import { UUID } from 'crypto';
 
-export function fetchSubstationPositions(studyUuid, currentNodeUuid, substationsIds) {
+export function fetchSubstationPositions(
+    studyUuid: UUID,
+    currentNodeUuid: UUID | undefined,
```
Comment (jamal-khey) [inline src/services/study/geo-data.ts:14]: i think it better to use '?' to make this argument optional
```suggestion
    currentNodeUuid?: UUID,
```

File: src/services/network-modification-types.ts [hunk 8]
```diff
@@ -37,3 +48,307 @@ export interface VscModificationInfo {
...
+    connectionName?: string | null;
+    connectionDirection?: string | null;
+    connectionPosition?: string | null;
+    terminalConnected?: boolean | null;
+    minP: number | null;
+    maxP: number | null;
+    isReactiveCapabilityCurveOn?: boolean;
+    minQ?: number | null;
+    maxQ?: number | null;
+    reactiveCapabilityCurve?: ReactiveCapabilityCurvePointsData | undefined;
```
Comment (souissimai) [inline src/services/network-modification-types.ts:71]: reactiveCapabilityCurve?: ReactiveCapabilityCurvePointsData ;
remove undefined

File: src/services/study/index.ts [hunk 9]
```diff
@@ -12,43 +12,67 @@ import {
...
 export const PREFIX_STUDY_QUERIES = import.meta.env.VITE_API_GATEWAY + '/study';
 
-export const getStudyUrl = (studyUuid) => `${PREFIX_STUDY_QUERIES}/v1/studies/${encodeURIComponent(studyUuid)}`;
+export const getStudyUrl = (studyUuid: UUID | null) =>
+    `${PREFIX_STUDY_QUERIES}/v1/studies/${safeEncodeURIComponent(studyUuid)}`;
 
-export const getStudyUrlWithNodeUuid = (studyUuid, nodeUuid) =>
-    `${PREFIX_STUDY_QUERIES}/v1/studies/${encodeURIComponent(studyUuid)}/nodes/${encodeURIComponent(nodeUuid)}`;
+export const getStudyUrlWithNodeUuid = (
+    studyUuid: UUID | string | null | undefined,
```
Comment (souissimai) [inline src/services/study/index.ts:30]: why? | string | null | undefined,

File: src/services/study/index.ts [hunk 10]
```diff
@@ -12,43 +12,67 @@ import {
...
+    withGeoData: boolean
+) {
     console.info(`Getting url of network area diagram of study '${studyUuid}' and node '${currentNodeUuid}'...`);
     return (
         getStudyUrlWithNodeUuid(studyUuid, currentNodeUuid) +
         '/network-area-diagram?' +
         new URLSearchParams({
-            depth: depth,
-            withGeoData: withGeoData,
+            depth: String(depth),
```
Comment (souissimai) [inline src/services/study/index.ts:61]: depth.toString()

File: src/services/study/loadflow.ts [hunk 11]
```diff
@@ -15,7 +25,7 @@ export function getDefaultLoadFlowProvider() {
     return backendFetchText(getDefaultLoadFlowProviderUrl);
 }
 
-export function setLoadFlowParameters(studyUuid, newParams) {
+export function setLoadFlowParameters(studyUuid: UUID, newParams: any) {
```
Comment (jamal-khey) [inline src/services/study/loadflow.ts:28]: any should be replaced by a type

File: src/services/study/network-modifications.ts [hunk 12]
```diff
@@ -28,21 +56,26 @@ export function changeNetworkModificationOrder(studyUuid, nodeUuid, itemUuid, be
     return backendFetch(url, { method: 'put' });
 }
 
-export function stashModifications(studyUuid, nodeUuid, modificationUuids) {
+export function stashModifications(studyUuid: UUID | null, nodeUuid: UUID | undefined, modificationUuids: UUID[]) {
     const urlSearchParams = new URLSearchParams();
-    urlSearchParams.append('stashed', true);
-    urlSearchParams.append('uuids', modificationUuids);
+    urlSearchParams.append('stashed', 'true');
```
Comment (jamal-khey) [inline src/services/study/network-modifications.ts:61]:
```suggestion
    urlSearchParams.append('stashed', String(true));
```

File: src/services/study/network-modifications.ts [hunk 13]
```diff
@@ -51,10 +84,10 @@ export function setModificationActivated(studyUuid, nodeUuid, modificationUuid,
     });
 }
 
-export function restoreModifications(studyUuid, nodeUuid, modificationUuids) {
+export function restoreModifications(studyUuid: UUID | null, nodeUuid: UUID | undefined, modificationUuids: UUID[]) {
     const urlSearchParams = new URLSearchParams();
-    urlSearchParams.append('stashed', false);
-    urlSearchParams.append('uuids', modificationUuids);
+    urlSearchParams.append('stashed', 'false');
```
Comment (jamal-khey) [inline src/services/study/network-modifications.ts:89]:
```suggestion
    urlSearchParams.append('stashed', String(false));
```

File: src/services/study/network-modifications.ts [hunk 14]
```diff
@@ -145,21 +196,25 @@ export function energiseEquipmentEnd(studyUuid, nodeUuid, branch, branchSide) {
...
-    substationsGeneratorsOrdering
+    studyUuid: UUID,
+    nodeUuid: UUID,
+    modificationUuid: UUID | undefined,
+    lossCoefficient: number,
+    defaultOutageRate: number,
+    generatorsWithoutOutage: any,
+    generatorsWithFixedActivePower: any,
+    generatorsFrequencyReserve: any,
+    substationsGeneratorsOrdering: any
```
Comment (jamal-khey) [inline src/services/study/network-modifications.ts:217]: you forgot those params ?

File: src/services/study/network-modifications.ts [hunk 15]
```diff
@@ -189,7 +244,13 @@ export function generationDispatch(
     });
 }
 
-export function generatorScaling(studyUuid, nodeUuid, modificationUuid, variationType, variations) {
+export function generatorScaling(
+    studyUuid: UUID,
+    nodeUuid: UUID,
+    modificationUuid: UUID | undefined,
+    variationType: string,
+    variations: any[]
```
Comment (jamal-khey) [inline src/services/study/network-modifications.ts:252]: any

File: src/services/study/network-modifications.ts [hunk 16]
```diff
@@ -982,36 +1045,36 @@ export function modifyLine(
...
+    ratedU1: number,
+    ratedU2: number,
+    currentLimit1: CurrentLimits,
+    currentLimit2: CurrentLimits,
+    voltageLevelId1: string,
+    busOrBusbarSectionId1: string,
+    voltageLevelId2: string,
+    busOrBusbarSectionId2: string,
+    ratioTapChanger: any,
+    phaseTapChanger: any,
```
Comment (jamal-khey) [inline src/services/study/network-modifications.ts:1066]: missing 2 types here

File: src/components/network-modification-tree-pane.jsx [hunk 17]
```diff
@@ -39,7 +39,7 @@ import {
     fetchNetworkModificationTreeNode,
     fetchStashedNodes,
 } from '../services/study/tree-subtree';
-import { buildNode, getUniqueNodeName, unbuildNode } from '../services/study';
+import { buildNode, getUniqueNodeName, unbuildNode } from '../services/study/index';
```
Comment (Tristan-WorkGH) [inline src/components/network-modification-tree-pane.jsx:42]:
```suggestion
import { buildNode, getUniqueNodeName, unbuildNode } from '../services/study';
```
"index" is always implicit in imports.

File: src/components/study-container.jsx [hunk 18]
```diff
@@ -36,7 +36,7 @@ import {
     connectNotificationsWsUpdateDirectories,
 } from '../services/directory-notification';
 import { useAllComputingStatus } from './computing-status/use-all-computing-status';
-import { fetchCaseName, fetchStudyExists } from '../services/study';
+import { fetchCaseName, fetchStudyExists } from '../services/study/index';
```
Comment (Tristan-WorkGH) [inline src/components/study-container.jsx:39]:
```suggestion
import { fetchCaseName, fetchStudyExists } from '../services/study';
```
same

File: src/redux/user-store.ts [hunk 19]
```diff
@@ -11,8 +11,8 @@
  */
 /*
     at src/redux/store.ts:12:33
```
Comment (Tristan-WorkGH) [inline src/redux/user-store.ts:13]: remove this debug comment, there no reason to have a outdated stacktrace in code

File: src/services/network-modification-types.ts [hunk 20]
```diff
@@ -5,7 +5,18 @@
...
  */
 
-import { ConverterStationElementModificationInfos } from '../components/dialogs/network-modifications/vsc/converter-station/converter-station-utils';
+import {
+    ConverterStationElementModificationInfos,
+    ReactiveCapabilityCurvePointsData,
+} from '../components/dialogs/network-modifications/vsc/converter-station/converter-station-utils';
+import { UUID } from 'crypto';
+import { Property } from '../components/dialogs/network-modifications/common/properties/property-utils';
+
```
Comment (Tristan-WorkGH) [inline src/services/network-modification-types.ts:14]:
```suggestion
```
no empty line in imports

File: src/services/network-modification-types.ts (hunk 3 above)
Comment (Tristan-WorkGH) [reply src/services/network-modification-types.ts:52]: and move it to a more generic place (for cyclic dependencies)

File: src/services/network-modification-types.ts [hunk 21]
```diff
@@ -37,3 +48,630 @@ export interface VscModificationInfo {
     connectionPosition?: string | null;
     terminalConnected?: boolean | null;
 }
+
+type AttributeModification<T> = { value: T; op: string };
+
+export interface BatteryModificationInfo {
```
Comment (Tristan-WorkGH) [inline src/services/network-modification-types.ts:54]: "field?: Type | null": can be `undefined` _and_ `null`?

File: src/services/study/loadflow.ts (hunk 11 above)
Comment (Tristan-WorkGH) [reply src/services/study/loadflow.ts:28]: or else `unknown` if we really don't know what it is.

File: src/services/study/network-modifications.ts (hunk 15 above)
Comment (AAJELLAL) [reply src/services/study/network-modifications.ts:259]: I couldn't find the variations type that's why I kept any as type.

File: src/services/study/network-modifications.ts (hunk 14 above)
Comment (AAJELLAL) [reply src/services/study/network-modifications.ts]: I don't quite understand the question.

File: src/services/study/network-modifications.ts (hunk 13 above)
Comment (AAJELLAL) [reply src/services/study/network-modifications.ts]: Done

File: src/services/study/network-modifications.ts (hunk 12 above)
Comment (AAJELLAL) [reply src/services/study/network-modifications.ts]: Done

File: src/services/study/loadflow.ts (hunk 11 above)
Comment (AAJELLAL) [reply src/services/study/loadflow.ts]: Done.

File: src/services/study/geo-data.ts (hunk 7 above)
Comment (AAJELLAL) [reply src/services/study/geo-data.ts]: Done

File: src/services/network-modification-types.ts (hunk 6 above)
Comment (AAJELLAL) [reply src/services/network-modification-types.ts]: Done

File: src/services/network-modification-types.ts (hunk 5 above)
Comment (AAJELLAL) [reply src/services/network-modification-types.ts]: Done.

File: src/services/network-modification-types.ts (hunk 4 above)
Comment (AAJELLAL) [reply src/services/network-modification-types.ts]: Done

File: src/services/network-modification-types.ts (hunk 3 above)
Comment (AAJELLAL) [reply src/services/network-modification-types.ts]: > converter-station-utils.tsx

Done.

File: src/services/directory.ts (hunk 2 above)
Comment (AAJELLAL) [reply src/services/directory.ts:13]: That's what it's used in the reducer..  [PARAM_FAVORITE_CONTINGENCY_LISTS]: UnknownArray;
Check the use of fetchContingencyAndFiltersLists.

File: src/components/network/network-map-tab.tsx (hunk 1 above)
Comment (AAJELLAL) [reply src/components/network/network-map-tab.tsx]: Done

File: src/services/network-modification-types.ts (hunk 8 above)
Comment (AAJELLAL) [reply src/services/network-modification-types.ts]: DOne

File: src/services/study/network-modifications.ts [hunk 22]
```diff
@@ -1211,7 +1274,7 @@ export function createSubstation(
  * Formats the properties of an array of properties so it can be consumed by the backend.
  * @returns {Array<{name: string, value: string, previousValue: string, added: boolean, deletionMark: boolean} | null>} - The modified properties.
  */
-export function formatPropertiesForBackend(previousProperties, newProperties) {
+export function formatPropertiesForBackend(previousProperties: any, newProperties: any) {
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/services/study/network-modifications.ts:1286]: types here seems to be :
{ name: string; value: string }[]
```suggestion
export function formatPropertiesForBackend(
    previousProperties: { name: string; value: string }[],
    newProperties: { name: string; value: string }[]
) {
```

[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/services/study/network-modifications.ts:1286]: no sorry try : Property defined in src/components/dialogs/network-modifications/common/properties/property-utils.ts

File: src/services/study/network-modifications.ts [hunk 23]
```diff
@@ -1221,7 +1284,7 @@ export function formatPropertiesForBackend(previousProperties, newProperties) {
     const previousPropertiesArray = Object.entries(previousProperties).map(([name, value]) => ({ name, value }));
     const newPropertiesArray = Object.entries(newProperties).map(([name, value]) => ({ name, value }));
 
-    const propertiesModifications = [];
+    const propertiesModifications: any = [];
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/services/study/network-modifications.ts:1296]: same Property[] in src/components/dialogs/network-modifications/common/properties/property-utils.ts

File: src/services/study/short-circuit-analysis.ts [hunk 24]
```diff
@@ -30,31 +67,37 @@ export function startShortCircuitAnalysis(studyUuid, currentNodeUuid, busId) {
...
+    type = ShortCircuitAnalysisType.ALL_BUSES
+) {
     const analysisType = getShortCircuitAnalysisTypeFromEnum(type);
     console.info(
         `Fetching ${analysisType} short circuit analysis status on '${studyUuid}' and node '${currentNodeUuid}' ...`
     );
     const urlSearchParams = new URLSearchParams();
-    urlSearchParams.append('type', analysisType);
+    if (analysisType !== null) {
+        urlSearchParams.append('type', analysisType);
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/services/study/short-circuit-analysis.ts:88]: I add this implementation based on other files. Check  fetchShortCircuitAnalysisPagedResults for example.

File: src/services/study/sensitivity-analysis.ts [hunk 25]
```diff
@@ -99,13 +148,21 @@ export function setSensitivityAnalysisParameters(studyUuid, newParams) {
...
+    isInjectionsSet: boolean,
+    newParams: SensitivityAnalysisFactorsCountParameters
+) {
     console.info('get sensitivity analysis parameters computing count');
     const urlSearchParams = new URLSearchParams();
     const jsoned = JSON.stringify(isInjectionsSet);
     urlSearchParams.append('isInjectionsSet', jsoned);
     Object.keys(newParams)
+        // @ts-ignore
+        //TODO: check this  later
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/services/study/sensitivity-analysis.ts]: Done.

File: src/components/network-modification-tree-pane.jsx (hunk 17 above)
Comment (AAJELLAL) [reply src/components/network-modification-tree-pane.jsx:42]: Done

File: src/services/study/sensitivity-analysis.ts [hunk 26]
```diff
@@ -28,14 +73,14 @@ export function stopSensitivityAnalysis(studyUuid, currentNodeUuid) {
...
-export function fetchSensitivityAnalysisStatus(studyUuid, currentNodeUuid) {
+export function fetchSensitivityAnalysisStatus(studyUuid: UUID, currentNodeUuid: UUID) {
     console.info(`Fetching sensitivity analysis status on ${studyUuid} and node ${currentNodeUuid} ...`);
     const url = `${getStudyUrlWithNodeUuid(studyUuid, currentNodeUuid)}/sensitivity-analysis/status`;
     console.debug(url);
     return backendFetchText(url);
 }
 
-export function fetchSensitivityAnalysisResult(studyUuid, currentNodeUuid, selector) {
+export function fetchSensitivityAnalysisResult(studyUuid: UUID, currentNodeUuid: UUID, selector: any) {
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/services/study/sensitivity-analysis.ts:59]: Done

File: src/services/study/network.ts [hunk 27]
```diff
@@ -277,7 +306,11 @@ export function fetchVoltageLevelsListInfos(studyUuid, currentNodeUuid, substati
     );
 }
 
-export function fetchVoltageLevelsMapInfos(studyUuid, currentNodeUuid, substationsIds) {
+export function fetchVoltageLevelsMapInfos(
+    studyUuid: UUID,
+    currentNodeUuid: UUID,
+    substationsIds: string[] | undefined
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/services/study/network.ts]: Done

File: src/services/study/network.ts [hunk 28]
```diff
@@ -266,7 +291,11 @@ export function fetchVoltageLevels(studyUuid, currentNodeUuid, substationsIds) {
     );
 }
 
-export function fetchVoltageLevelsListInfos(studyUuid, currentNodeUuid, substationsIds) {
+export function fetchVoltageLevelsListInfos(
+    studyUuid: UUID,
+    currentNodeUuid: UUID,
+    substationsIds?: string[] | undefined
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/services/study/network.ts]: Done

File: src/services/study/network.ts [hunk 29]
```diff
@@ -255,7 +280,7 @@ export function fetchLines(studyUuid, currentNodeUuid, substationsIds) {
     );
 }
 
-export function fetchVoltageLevels(studyUuid, currentNodeUuid, substationsIds) {
+export function fetchVoltageLevels(studyUuid: UUID, currentNodeUuid: UUID, substationsIds: string[] | undefined) {
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/services/study/network.ts]: Done

File: src/services/study/network.ts [hunk 30]
```diff
@@ -244,7 +269,7 @@ export function fetchSubstations(studyUuid, currentNodeUuid, substationsIds) {
     );
 }
 
-export function fetchLines(studyUuid, currentNodeUuid, substationsIds) {
+export function fetchLines(studyUuid: UUID, currentNodeUuid: UUID, substationsIds: string[] | undefined) {
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/services/study/network.ts]: Done

File: src/services/study/network.ts [hunk 31]
```diff
@@ -233,7 +258,7 @@ export function fetchHvdcLinesMapInfos(studyUuid, currentNodeUuid, substationsId
     );
 }
 
-export function fetchSubstations(studyUuid, currentNodeUuid, substationsIds) {
+export function fetchSubstations(studyUuid: UUID, currentNodeUuid: UUID, substationsIds: string[] | undefined) {
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/services/study/network.ts]: Done

File: src/services/study/network-modifications.ts [hunk 32]
```diff
@@ -1845,11 +1945,18 @@ export function modifyByFormula(studyUuid, nodeUuid, equipmentType, formulas, is
...
+    equipmentType: string,
+    assignmentsList: Assignment[],
+    isUpdate: boolean,
+    modificationUuid: UUID | null
+) {
     let modificationUrl = getNetworkModificationUrl(studyUuid, nodeUuid);
 
     if (isUpdate) {
-        modificationUrl += '/' + encodeURIComponent(modificationUuid);
+        modificationUrl += '/' + safeEncodeURIComponent(modificationUuid);
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/services/study/network-modifications.ts:2016]: As we discussed at the daily , based on what it is done before the studyUuid and the modificationUuid could be null or undefined while loading the app. But when the call is actually done, they would never be null.
If I add the check before, what should be the behaviour?

File: src/components/dialogs/parameters/short-circuit-parameters.tsx [hunk 33]
```diff
@@ -104,15 +104,15 @@ export const useGetShortCircuitParameters = (): UseGetShortCircuitParametersProp
...
+        [SHORT_CIRCUIT_WITH_FEEDER_RESULT]: yup.boolean().required(),
         [SHORT_CIRCUIT_PREDEFINED_PARAMS]: yup
             .mixed<PREDEFINED_PARAMETERS>()
             .oneOf(Object.values(PREDEFINED_PARAMETERS))
             .required(),
-        [SHORT_CIRCUIT_WITH_LOADS]: yup.boolean(),
-        [SHORT_CIRCUIT_WITH_VSC_CONVERTER_STATIONS]: yup.boolean(),
-        [SHORT_CIRCUIT_WITH_SHUNT_COMPENSATORS]: yup.boolean(),
-        [SHORT_CIRCUIT_WITH_NEUTRAL_POSITION]: yup.boolean(),
+        [SHORT_CIRCUIT_WITH_LOADS]: yup.boolean().required(),
```
Comment (jamal-khey) [inline src/components/dialogs/parameters/short-circuit-parameters.tsx:112]: why the attributes are required now ?

**Review by jamal-khey — COMMENTED** [review]
looks good

**Review by jamal-khey — APPROVED** [review]
things can be improved but it's okey to merge it as it is . so we avoid future conflicts.
Test Ok

**Review by souissimai — APPROVED** [review]
tested

**Pull request merged**